### PR TITLE
Add a note to link JIRA issues to PRs

### PR DESCRIPTION
**Pull request by meichstedt**
## High Level Description

This affects the pull request template and asks the pull request creator to add a link to the related JIRA issue requesting the bump.

## Related Issues

no related issues

## Checklist for all PR's

n/a

## Checklist for component/package updates:

n/a
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by meichstedt** [comment]
@mesosphere-mergebot label Ready For Review

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 What features does this change enable? What bugs does this change fix? High-Level description of how things changed.
 
+## Related JIRA
+
+Please link the JIRA issue requiring this bump so it can be closed once the PR is merged.
```
Comment (adam-mesos) [inline PULL_REQUEST_TEMPLATE.md]: I thought that's what the "Related Issues" section below is for. What's the difference?

Comment (marcomonaco) [reply PULL_REQUEST_TEMPLATE.md]: The reason for this that we want to have a dedicated section to link the corresponding JIRA bump ticket. For example, the Related JIRA is a new Marathon version bump, and the Related tickets are any JIRA tickets which have been resolved with this bump.

Comment (orlandohohmeier) [reply PULL_REQUEST_TEMPLATE.md]: We probably should run with a different title to clearly state the difference between these two sections. How about `Associated Issues` of `Linked Issues`?

```markdown

## Associated Issues

<!-- Please link all issue associated (e.g., bump or release tracking issues) with this PR. --> 

- [DCOS_OSS-<number>](https://jira.dcos.io/browse/DCOS_OSS-<number>) Update Foo and Bar. 


## Related Issues

<!-- Please link all related issue (e.g., tasks and bugs) addressed with this PR. --> 

- [DCOS_OSS-<number>](https://jira.dcos.io/browse/DCOS_OSS-<number>) Foo the Bar so it stops Bazzing.

**Comment by jgehrcke** [comment]
I triggered this effort and would like to write down the goal in my words. IMO we need to have a section like this:

> *IF* there is/are (a) JIRA ticket(s) that you want to have closed, blindly, right after landing *that* PR, then please link it here: ...

This is a subset of the "Related issues" set. It will vastly simplify the work of closing tickets right after merging. And my hope is that it also forces us a little bit into the direction where every PR has at least such a ticket associated. Which can at some point be used for auto-closing upon merge.

**Comment by orsenthil** [comment]
We already request for JIRA for every PR in the template. The problem is with our enforcement and common understanding on what JIRA to provide.

Given that, I don't see very high value in this PR. I am okay either being merged/rejected. Both seem same to me.

>  And my hope is that it also forces us a little bit into the direction where every PR has at least such a ticket associated
  
+1 to this. I am contemplating for while on how to bring enforcement. One idea I have is, the title of the PR should start with JIRA ticket and mergebot status check can enforce that.

And once mergebot has information on JIRA associated with the PR, it could update the JIRA when the PR is merged. 

If there are mechanisms of JIRA+Github workflow outside of we writing some integration, I am interested in knowing that too.

**Comment by jgehrcke** [comment]
Ship-it was applied prematurely. See DCOS-19252.

**Comment by jgehrcke** [comment]
That's the situation I'd like to address: https://github.com/dcos/dcos/pull/2045#issue-268283154 -- this happens rather often. After merging a patch, it's unclear how to reflect that in JIRA. We lack the discipline. Precise wording will IMO help.

**Comment by jgehrcke** [comment]
I took liberty and used the insights from this discussion to create https://github.com/dcos/dcos/pull/2051 which is now meant to supersede this PR.

**Comment by meichstedt** [comment]
Thanks @jgehrcke I'll close this one then!

**Pull request closed without merging**